Objective-C/Internal/Replicator/CBLWebSocket.mm

```diff
@@ -538,6 +543,12 @@ - (void) didCloseWithError: (NSError*)error {
 }
 
 
+- (void) closeTask {
+    [_task closeWrite];
+    [_task closeRead];
+}
```

Comment:
Should you set `_task = nil` here, since it's now closed? Then you don't have to set it to nil after the various calls to -closeTask.

Reply:
I tried to do this, but the changing scope is getting bigger. There is some complication especially with -didCloseWithError: that shouldn't be called when _task is nil. I think we should do this in post 2.0 after we have enough time to test the change to cover all edge cases. I put todo in the code.